server/insights/src/main/java/org/infinispan/server/insights/report/InfinispanSubreport.java

```diff
@@ -0,0 +1,31 @@
+package org.infinispan.server.insights.report;
+
+import org.infinispan.manager.CacheManagerInfo;
+
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.redhat.insights.InsightsSubreport;
+
+public class InfinispanSubreport implements InsightsSubreport {
+
+   private final CacheManagerInfo cacheManagerInfo;
+   String jsonReport;
+
+   public InfinispanSubreport(CacheManagerInfo cacheManagerInfo) {
+      this.cacheManagerInfo = cacheManagerInfo;
+   }
+
+   @Override
+   public void generateReport() {
+      jsonReport = cacheManagerInfo.toJson().toString();
```

Comment:
Seems it would be better to store the Json representation instead of the String? Then the serialization can just pass the Json representation directly instead of using raw.

Reply:
Do you mean to pass the Json representation directly to `com.fasterxml.jackson.core.JsonGenerator` instead of calling `com.fasterxml.jackson.core.JsonGenerator#writeRaw(String)`?
If so, how can I do it?

Reply:
We would want to add a utility to do it, but it is nothing flashy just some some nice recursive calls checking what level the json is and invoking the appropriate `write***` method on the JsonGenerator class based on the Json types we have.

e.g.

```java
Json json = cacheManagerInfo.toJson();

recursiveWrite(json, jsonGenerator);
```
```java
void recursiveWrite(Json json, JsonGenerator generator) {
   if (json.isObject()) {
      generator.startObject();
      var map = json.asJsonMap();
      for (Entry<String, Json> entry : map.entrySet()) {
         generator.writeFieldName(entry.getKey());
         recursiveWrite(entry.getValue(), generator);
      }
      generator.writeEndObject();
   } else if (
   .... rest
}
```

But I guess since this isn't a performance required operation, we can just stick with what you have.

Reply:
I think `CacheManagerInfo` doesn't contain all the information we want. Without exposing specifics, we could use the `CacheResourceV2.CacheFullDetail` information which gives us high-level info such as `indexed`, `secure`, `transactional`, etc

Reply:
Additionally, we should not be including "private" information in the report, such as IP addresses, hostnames, etc